sharding/indexHashedNodesCoordinator.go

```diff
@@ -591,6 +627,56 @@ func (ihgs *indexHashedNodesCoordinator) GetConsensusWhitelistedNodes(
 	return shardEligible, nil
 }
 
+// UpdatePeersListAndIndex will update the list and the index for all peers
+func (ihgs *indexHashedNodesCoordinator) UpdatePeersListAndIndex() error {
+	ihgs.mutNodesConfig.RLock()
+	defer ihgs.mutNodesConfig.RUnlock()
+
+	nodesConfig, ok := ihgs.nodesConfig[ihgs.currentEpoch]
```

Comment:
I don't think it would be ok because in the inner called functions we use the nodesConfig variable which has the same pointer as the element in the map. I would leave it like this

Reply:
I don't think it would be ok because in the inner called functions we use the nodesConfig variable which has the same pointer as the element in the map. I would leave it like this

Reply:
That is not an issue, the pointer can change inside the original map, but you already have a copy of the pointer so it is fine.

Inside the nodesCofig you now hold, when accessing the eligible and waiting maps you have other mutexes.